src/main/java/practice/StreamPractice.java

```
@@ -23,7 +34,11 @@ public int findMinEvenNumber(List<String> numbers) {
      * But before that subtract 1 from each element on an odd position (having the odd index).
      */
     public Double getOddNumsAverage(List<Integer> numbers) {
-        return 0D;
+        return IntStream.range(0, numbers.size())
+                .map(i -> (i % 2 != 0) ? numbers.get(i) - 1 : numbers.get(i))
+                .filter(n -> n % 2 != 0)
+                .average()
+                .orElseThrow(NoSuchElementException::new);
```

The use of 'get()' or 'getAsDouble()' is recommended when you need to throw an exception if no result is found after stream execution. In this case, replacing 'orElseThrow' with 'getAsDouble()' would be more appropriate, and the exception should be specific to the context, not a generic NoSuchElementException.